static/canvas.js

```diff
@@ -126,8 +128,27 @@ class Canvas {
         /* Event for moving the mouse. Primarily used for enlarging a box and moving it. */
         canvas.addEventListener('mousemove', function (e) {
             var mouse = myState.getMouse(e);
-            if (!myState.enlargeDirection && !myState.dragging) {
-                return;
+            var mx = mouse.x;
+            var my = mouse.y;
+            var boxes = myState.boxes;
+            var l = boxes.length;
+            var border = false;
+            var box = null;
+
+            /* Checks if mouse is hovering over a border. */
+            for (var i = l - 1; i >= 0; i--) {
+                box = boxes[i]; 
+                border = boxes[i].withinBorder(mx, my);
+                if (border) {
+                    break;
+                }
+            }
+
+
+            if (!myState.enlargeDirection && !myState.dragging && !border) {
+                document.body.style.cursor = 'default';
+            } else if (!myState.enlargeDirection && !myState.dragging && border) {
+                myState.doubleArrow(box, mx, my);
```

Comment:
Yeah, this wasn't really a problem for the sides of a box so much as it was for corners, since a corner's mouse style can change depending on how you drag. Also I had to have extra conditionals for the sides of a cursor because I realized the withinBorder function in box.js specified only either the y or x coordinate for a border. For example if it dealt with the top border, it would only specify the y coordinates for that border, meaning if I used this to change the cursor style, the cursor would remain a double sided arrow when it was along that y coordinate, even if it wasn't hovering over a box.